src/tensorlake/function_executor/service.py

```diff
@@ -256,3 +357,159 @@ def get_info(
         self, request: InfoRequest, context: grpc.ServicerContext
     ) -> InfoResponse:
         return InfoResponse(**info_response_kv_args())
+
+    def _execute_task_in_thread(self, task_info: _TaskInfo):
+        def progress_reporter(progress: Progress):
+            with task_info.execution.updated:
+                task_info.execution.progress = ProgressUpdate(
+                    current=progress.current, total=progress.total
+                )
+                task_info.execution.updated.notify_all()
+            # sleep(0) here momentarily releases the GIL, giving other
+            # threads a chance to run - e.g. allowing the FE to handle
+            # incoming RPCs, to report back await_task() progress
+            # messages, &c.
+            time.sleep(0)
+
+        try:
+            # Run the task handler
+            result = RunTaskHandler(
+                task=task_info.task,
+                invocation_state=ProxiedInvocationState(
+                    task_info.task.task_id, self._invocation_state_proxy_server
+                ),
+                function_wrapper=self._function_wrapper,
+                function_stdout=self._function_stdout,
+                function_stderr=self._function_stderr,
+                graph_metadata=self._graph_metadata,
+                progress_reporter=progress_reporter,
+                logger=self._logger,
+            ).run()
+
+            task_info.task.result.CopyFrom(result)
+
+        except BaseException as e:
+            # Handle any errors by creating a failed task result
+            self._logger.error(
+                "task execution failed in background thread",
+                task_id=task_info.task.task_id,
+                exc_info=e,
```

Comment:
log tags are missing (invocation_id, allocation_id), see the run task handler logger. task_info.logger would do the job and it can be passed around with task_info to other code paths. We can gradually migrate them to use it.

Reply:
SGTM